oat-sa/tao-core
# Feature/AUT-3632/Apply the Solar Design to the forms (step 2)

### Related to: https://oat-sa.atlassian.net/browse/AUT-3646

### Summary

Apply the Solar Design to the forms
![image](https://github.com/oat-sa/tao-core/assets/1500098/10ea6046-f957-44d3-89ae-2eb51ce6f78b)
![image](https://github.com/oat-sa/tao-core/assets/1500098/4c32fe6e-3cc6-414d-bd24-dd77666af4c1)
![image](https://github.com/oat-sa/tao-core/assets/1500098/8754ed83-0641-460e-9197-3c641d923543)
![image](https://github.com/oat-sa/tao-core/assets/1500098/61d99437-b33c-42ce-8386-6d4f0a97ac6c)


### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

The forms are now following the Solar Design.

TreeViews has been refactored to also style the resource selector.

Common style sets are moved to mixins for easier reuse.


### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3632/solar-design-forms-step-2`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the forms, compare with the UX mockups
